benchmark/url/urlpattern-parse.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+const common = require('../common.js');
+const { URLPattern } = require('url');
+
+const tests = [
+  'https://(sub.)?example(.com/)foo',
+  { 'hostname': 'xn--caf-dma.com' },
+  { 'pathname': '/foo', 'search': 'bar', 'hash': 'baz',
+    'baseURL': 'https://example.com:8080' },
+  { 'pathname': '/([[a-z]--a])' },
+];
+
+const bench = common.createBenchmark(main, {
+  pattern: tests.map(JSON.stringify),
+  n: [1e5],
+});
+
+function main({ pattern, n }) {
+  const inputPattern = JSON.parse(pattern);
+  bench.start();
+  for (let i = 0; i < n; i += 1) {
+    new URLPattern(inputPattern);
```

Comment:
It's a good idea to assign this to a variable, and then later do a `assert.ok` just to try to avoid dead-code elimination by v8.

Reply:
done